bpf-gpl/conntrack.h

```diff
@@ -72,8 +72,12 @@ struct calico_ct_leg {
 	__u32 whitelisted:1;
 
 	__u32 opener:1;
+
+	__u32 ifindex; /* where the packet entered the system from */
 };
 
+#define CT_INVALID_IFINDEX	((__u32) -1)
```

Comment:
From what I can tell, 0 is not a valid ifIndex (but all other uint values are) so why not use 0 for the invalid value?

Reply:
https://elixir.bootlin.com/linux/latest/source/include/net/flow.h#L18

Reply:
though that 0 is valid, will change it, good catch